src/sinks/uds.rs

```diff
@@ -55,28 +56,25 @@ impl UdsMetricSink {
     ///
     /// # Non-blocking Example
     ///
-    /// Note that putting the UDS socket into non-blocking mode is the
-    /// default when sink and socket are automatically created with the
-    /// `StatsdClient::from_uds_path` method.
-    ///
     /// ```no_run
-    /// use std::os::unix::net::UnixStream;
+    /// use std::os::unix::net::UnixDatagram;
     /// use cadence::UdsMetricSink;
     ///
-    /// let socket = UnixStream::connect("/tmp/sock").unwrap();
+    /// let socket = UnixDatagram::unbound().unwrap();
+    /// socket.connect("/tmp/sock").unwrap();
     /// socket.set_nonblocking(true).unwrap();
     /// let sink = UdsMetricSink::from(socket);
     /// ```
-    pub fn from(socket: UnixStream) -> UdsMetricSink {
-        UdsMetricSink {
-            socket: Mutex::new(socket),
-        }
+    pub fn from(socket: UnixDatagram) -> UdsMetricSink {
```

Comment:
Can this be modified to have similar interface to the UDP one, i.e. take a path and a socket as arguments? Then, in the `.emit()` method below you'd call `self.socket.send_to(bytes, path)`.

Reply:
Done. I use `P: AsRef<Path>` to be consistent with the underlying API, e.g. https://doc.rust-lang.org/std/os/unix/net/struct.UnixDatagram.html#method.bind.

Reply:
:+1: